### PR TITLE
[Function][Vectorized] Implement ifnull vectorized function

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_ifnull.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+class FunctionIfNull : public IFunction {
+public:
+    static constexpr auto name = "ifnull";
+
+    static FunctionPtr create() { return std::make_shared<FunctionIfNull>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 2; }
+    //TODO:
```
Comment (HappenLee) [inline be/src/vec/functions/function_ifnull.h:37]: 这个TODO是？

File: be/src/vec/functions/function_ifnull.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            block.get_by_position(result),
+        };
+        const ColumnWithTypeAndName new_column {
+            col_left,
+        };
+        /// compute isnull(col_left)
+        if (auto* nullable = check_and_get_column<ColumnNullable>(*col_left.column)) {
+            block.get_by_position(arguments[0]).column = nullable->get_null_map_column_ptr();
+        } else {
+            block.get_by_position(arguments[0]).column = col_left.column;
```
Comment (HappenLee) [inline be/src/vec/functions/function_ifnull.h:61]: 这里应该是result的column吧，这里column要拷贝一下，以便进行mem_reuse

**Review by HappenLee — COMMENTED** [review]
辛苦添加一下ut

**Comment by xinghuayu007** [comment]
> 辛苦添加一下ut

已添加

**Pull request merged**